case-lib/lib.sh

```diff
@@ -40,6 +40,97 @@ if [ ! "$SOFCARD" ]; then
 		awk '/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1; exit 0;}')
 fi
 
+_SOF_TEST_LOCK_FILE=/tmp/sof-test-card"$SOFCARD".lock
+
+start_test()
+{
+    if is_subtest; then
+        return 0
+    fi
+
+    test -z "$SOF_TEST_TOP_PID" || {
+        dlogw "SOF_TEST_TOP_PID=$SOF_TEST_TOP_PID already defined, multiple lib.sh inclusions?"
+        return 0
+    }
+
+    export SOF_TEST_TOP_PID="$$"
+    local prefix; prefix="ktime=$(ktime) sof-test PID=$SOF_TEST_TOP_PID"
+    local ftemp; ftemp=$(mktemp --tmpdir sof-test-XXXXX)
+    printf '%s' "$SOF_TEST_TOP_PID" > "$ftemp"
+
+    # `ln` is supposedly atomic. `mv --no-clobber` fails with... exit
+    # status 0! It's not even clear whether it's atomic.
+    ln "$ftemp" "${_SOF_TEST_LOCK_FILE}" || {
+        local lock_pid; lock_pid=$(head -n 1 "${_SOF_TEST_LOCK_FILE}")
+
+        if [ "$SOF_TEST_TOP_PID" = "$lock_pid" ]; then
+            # Internal error
+            die "${_SOF_TEST_LOCK_FILE} with ${SOF_TEST_TOP_PID} already exists?!"
+        fi
+
+        # Assume this was left-over after a crash, keep running.
+        # If another test is really running concurrently then stealing
+        # the lock maximizes the changes of BOTH failing.
```

Comment:
> changes = chance?

Thanks, good catch.

> Also, is there a reason you use curly braces on line 77 but not 76?

No particular reason. The inconsistency from one line to the next is indeed ugly. Braces are unnecessary here, I'll remove them.

AFAIK braces are strictly required only in situations like these:
```
varA="a"
varAA="aa"
echo "$varAA"
echo "${varA}A"

```
Otherwise they're only for readibility / style / consistency.
https://google.github.io/styleguide/shellguide.html#variable-expansion